quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesRequiredCodeActionTest.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.quarkus.services;
+
+import static com.redhat.quarkus.services.QuarkusAssert.ca;
+import static com.redhat.quarkus.services.QuarkusAssert.d;
+import static com.redhat.quarkus.services.QuarkusAssert.te;
+import static com.redhat.quarkus.services.QuarkusAssert.testCodeActionsFor;
+import static com.redhat.quarkus.services.QuarkusAssert.testDiagnosticsFor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.redhat.quarkus.commons.ExtendedConfigDescriptionBuildItem;
+import com.redhat.quarkus.commons.QuarkusProjectInfo;
+import com.redhat.quarkus.ls.commons.BadLocationException;
+import com.redhat.quarkus.settings.QuarkusFormattingSettings;
+import com.redhat.quarkus.settings.QuarkusValidationSettings;
+import com.redhat.quarkus.settings.QuarkusValidationTypeSettings;
+
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test with the required code actions in 'application.properties' file.
+ * 
+ *
+ */
+public class ApplicationPropertiesRequiredCodeActionTest {
+
+	private static QuarkusProjectInfo projectInfo;
+	private static QuarkusValidationSettings validationSettings;
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		projectInfo = new QuarkusProjectInfo();
+		validationSettings = new QuarkusValidationSettings();
+		List<ExtendedConfigDescriptionBuildItem> properties = new ArrayList<ExtendedConfigDescriptionBuildItem>();
+		ExtendedConfigDescriptionBuildItem p1 = new ExtendedConfigDescriptionBuildItem();
+		p1.setPropertyName("quarkus.required.property");
+		p1.setRequired(true);
+		properties.add(p1);
+		ExtendedConfigDescriptionBuildItem p2 = new ExtendedConfigDescriptionBuildItem();
+		p2.setPropertyName("quarkus.second.required.property");
+		p2.setRequired(true);
+		properties.add(p2);
+		ExtendedConfigDescriptionBuildItem p3 = new ExtendedConfigDescriptionBuildItem();
+		p3.setPropertyName("quarkus.third.required.property");
+		p3.setRequired(true);
+		properties.add(p3);
+		ExtendedConfigDescriptionBuildItem p4 = new ExtendedConfigDescriptionBuildItem();
+		p4.setPropertyName("quarkus.optional.property");
+		p4.setRequired(false);
+		properties.add(p4);
+		projectInfo.setProperties(properties);
+
+		QuarkusValidationTypeSettings error = new QuarkusValidationTypeSettings();
+		error.setSeverity("warning");
+		validationSettings.setRequired(error);
+	}
+
+	@Test
+	public void testOneMissingPropertyCodeAction() throws BadLocationException {
```

Comment:
Copy/paste this test and do the same test but with `\r\n` to check the generated code action uses the proper line delimiter.

Reply:
done